### PR TITLE
sceFiber Windows implementation

### PR DESCRIPTION
**Pull request by mailwl**

### CODE REVIEW

File: GPCS4/SceModules/SceFiber/sce_fiber.cpp [hunk 1]
```diff
@@ -1,26 +1,58 @@
...
+
+	return SCE_OK;
+}
 
-int PS4API sceFiberFinalize(void)
+int PS4API sceFiberFinalize(SceFiber* fiber)
 {
 	LOG_FIXME("Not implemented");
+	UtilFiber::FinalizeFiber(fiber->fiber);
+	delete fiber;
```
Comment (Inori) [inline GPCS4/SceModules/SceFiber/sce_fiber.cpp]: delete inside `UtilFiber::FinalizeFiber`

File: GPCS4/SceModules/SceFiber/sce_fiber.cpp [hunk 2]
```diff
@@ -32,25 +64,52 @@ int PS4API sceFiberReturnToThread(void)
...
+	
 	return SCE_OK;
 }
 
-int PS4API sceFiberGetSelf(void)
+
+int PS4API sceFiberGetSelf(SceFiber** fiber)
 {
 	LOG_FIXME("Not implemented");
+	*fiber = currentFiber;
```
Comment (Inori) [inline GPCS4/SceModules/SceFiber/sce_fiber.cpp]: can we implement this using `GetCurrentFiber`

File: GPCS4/SceModules/SceFiber/sce_fiber.h [hunk 3]
```diff
@@ -19,29 +19,44 @@ extern const SCE_EXPORT_MODULE g_ExpModuleSceFiber;
 // The codebase is generated using GenerateCode.py
 // You may need to modify the code manually to fit development needs
 
+typedef void PS4API (*SceFiberEntry)(uint64_t argOnInitialize, uint64_t argOnRun) __attribute__((noreturn));
```
Comment (Inori) [inline GPCS4/SceModules/SceFiber/sce_fiber.h]: these's a `PS4NORETURN` for `__attribute__((noreturn))`
see:
https://github.com/Inori/GPCS4/blob/master/GPCS4/Common/GPCS4Decoration.h

File: GPCS4/SceModules/SceFiber/sce_fiber.cpp [hunk 4]
```diff
@@ -32,25 +64,52 @@ int PS4API sceFiberReturnToThread(void)
 }
 
 
-int PS4API sceFiberRun(void)
+int PS4API sceFiberRun(SceFiber* fiber, uint64_t argOnRunTo, uint64_t* argOnReturn)
 {
 	LOG_FIXME("Not implemented");
+	fiber->fiber = UtilFiber::CreateAndRunFiber(fiber->entry, fiber->argOnInitialize, argOnRunTo);
```
Comment (Inori) [inline GPCS4/SceModules/SceFiber/sce_fiber.cpp]: I think we should initialize the primary fiber in `_sceFiberInitializeImpl`

File: GPCS4/SceModules/SceFiber/sce_fiber.cpp (hunk 2 above)
Comment (mailwl) [reply GPCS4/SceModules/SceFiber/sce_fiber.cpp:98]: GetCurrentFiber returns Windows Fiber, but we need SceFiber. Probably we need vector of created sceFibers and search current windows fiber in this vector

Comment (Inori) [reply GPCS4/SceModules/SceFiber/sce_fiber.cpp:99]: No, there's no difference by using a global `vector` compared to a global variable.
We can set the `SceFiber` in the fiber's local storage during `CreateFiber` by passing `lpParameter`, then use `GetFiberData` to retrieve the data.

**Pull request closed without merging**